kingpin/actors/aws/s3.py

```diff
@@ -162,10 +183,23 @@ def _create_bucket(self):
         returns:
             <A boto.s3.Bucket object>
         """
+        # If we're running in DRY mode, then we create a fake bucket object
+        # that will be passed back. This mock object lets us simplify the rest
+        # of our code because we can mock out the results of creating a fresh
+        # empty bucket with no policies, versions, etc.
         if self._dry:
             self.log.warning('Would have created s3://%s' %
                              self.option('name'))
-            raise gen.Return()
+
+            # Generate a fake bucket and return it
+            mock_bucket = mock.MagicMock(name=self.option('name'))
```

Comment:
I would suggest, use moto to mock boto when in dry, then you don't have to manually create this mock.  You could also use a decorator that would look at the --dry flag and mock boto when true.